clippy_lints/src/len_zero.rs

```diff
@@ -192,7 +192,7 @@ impl<'tcx> LateLintPass<'tcx> for LenZero {
 
         if let ExprKind::Binary(Spanned { node: cmp, .. }, left, right) = expr.kind {
             // expr.span might contains parenthesis, see issue #10529
-            let actual_span = left.span.with_hi(right.span.hi());
```

Comment:
this "hack" does not work with macros very well, the left and right spans in the following example are:

```rust
(compare_to!(0) < has_is_empty.len()).then(|| println!("This can happen."));
//              lhs:  -
//                      rhs: -------------------------- 
```

which causing an `)` being included in the `actual_span`... So I change this back to using the binary expr span but strip the parentheses. However the `SourceMap::start_point` method seems expensive at first glimpse, I'm still exploring more optimized ways for this

Reply:
Just ran speedtest with 100x iteration for this pr, the result was:

```
average UI time: 215 millis.
```

and on master branch:

```
average UI time: 216 millis.
```

lol :rofl: , I guess I shouldn't worry too much about the performance...